### PR TITLE
Return Blank for blank input of Text

### PR DESCRIPTION
**Pull request by nguhoa**
Fix issue: https://github.com/microsoft/Power-Fx/issues/1458

### CODE REVIEW

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Text.txt [hunk 1]
```diff
@@ -5,7 +5,7 @@
 "123.400"
 
 >> Text(Blank(), "#.000")
-""
+Blank()
```
Comment (CarlosFigueiraMSFT) [inline src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Text.txt:8]: In the latest discussions with PMs we agreed that if there is only one argument to the Text function (i.e., it's being used as a conversion function) then blank values are propagated. However, if we explicitly ask for a format, as in this example, then the result should be an empty string (`""`)

File: src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Text.txt [hunk 2]
```diff
@@ -136,32 +136,37 @@
...
 "pqr"
 
-
 >> Text(1/0, "#.000")
 Error({Kind:ErrorKind.Div0})
 
+>> Text(If(1<0,2))
+Blank()
+
+>> Text(ParseJSON("null"))
```
Comment (CarlosFigueiraMSFT) [inline src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Text.txt:167]: We already have this test in ParseJson.txt, no need to repeat it

File: src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Text.txt (hunk 1 above)
Comment (nguhoa) [reply src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Text.txt]: Updated to handle this. It was not clear in the problem. Thanks for explaining here.

File: src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Text.txt (hunk 2 above)
Comment (nguhoa) [reply src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Text.txt]: remove this.

File: src/libraries/Microsoft.PowerFx.Interpreter/Functions/StandardErrorHandling.cs [hunk 3]
```diff
@@ -18,7 +18,8 @@ private enum ReturnBehavior
             AlwaysEvaluateAndReturnResult,
             ReturnBlankIfAnyArgIsBlank,
             ReturnEmptyStringIfAnyArgIsBlank,
-            ReturnFalseIfAnyArgIsBlank
+            ReturnFalseIfAnyArgIsBlank,
+            ReturnBlankIfOneArgIsBlankAndEmptyStringForOtherArgIsBlank
```
Comment (CarlosFigueiraMSFT) [inline src/libraries/Microsoft.PowerFx.Interpreter/Functions/StandardErrorHandling.cs:22]: >ReturnBlankIfOneArgIsBlankAndEmptyStringForOtherArgIsBlank [](http://example.com/codeflow?start=12&length=58)

This is way too specific - it's only used by the Text function. Do we need to have a new one, or should just use AlwaysEvaluateAndReturnResult and let the Text function handle it?

Comment (nguhoa) [reply src/libraries/Microsoft.PowerFx.Interpreter/Functions/StandardErrorHandling.cs]: Updated here and handle in text.

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryText.cs [hunk 4]
```diff
@@ -295,7 +295,20 @@ public static bool TryDecimal(FormattingInfo formatInfo, IRContext irContext, Fo
 
         // https://docs.microsoft.com/en-us/powerapps/maker/canvas-apps/functions/function-text
         public static FormulaValue Text(EvalVisitor runner, EvalVisitorContext context, IRContext irContext, FormulaValue[] args)
-        {
+        {
+            if (args.Length == 1 && (args[0] is BlankValue || (args[0] is UntypedObjectValue untypedObjectValue && untypedObjectValue.Impl.Type == FormulaType.Blank)))
+            {
+                return new BlankValue(irContext);
```
Comment (CarlosFigueiraMSFT) [inline src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryText.cs:301]: Can you add a comment explaining the logic? Something like "// When used as a pure conversion function (single argument, no format string), Text() propagates null values"

File: src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryText.cs [hunk 5]
```diff
@@ -295,7 +295,20 @@ public static bool TryDecimal(FormattingInfo formatInfo, IRContext irContext, Fo
 
         // https://docs.microsoft.com/en-us/powerapps/maker/canvas-apps/functions/function-text
         public static FormulaValue Text(EvalVisitor runner, EvalVisitorContext context, IRContext irContext, FormulaValue[] args)
-        {
+        {
+            if (args.Length == 1 && (args[0] is BlankValue || (args[0] is UntypedObjectValue untypedObjectValue && untypedObjectValue.Impl.Type == FormulaType.Blank)))
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryText.cs:299]: >args[0] is BlankValue || (args[0] is UntypedObjectValue untypedObjectValue && untypedObjectValue.Impl.Type == FormulaType.Blank [](http://example.com/codeflow?start=37&length=127)

Do we have other places that need this check?

I wonder if we should encapsulate it in a new helper like:
FormulaValue.IsBlank()

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryText.cs (hunk 4 above)
Comment (nguhoa) [reply src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryText.cs:302]: Add comment. thanks

File: src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryText.cs (hunk 5 above)
Comment (nguhoa) [reply src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryText.cs]: Yes, we use in few places so I updated Greg's isblank function to share.

**Review by MikeStall — APPROVED** [review]
:shipit:

**Comment by LucGenetier** [comment]
✅ No public API change.

**Comment by LucGenetier** [comment]
✅ No public API change.

**Pull request merged**